raiden/transfer/log.py

```diff
@@ -83,6 +84,12 @@ def __init__(self, database_path):
             'FOREIGN KEY(statechange_id) REFERENCES state_changes(id)'
             ')'
         )
+        cursor.execute(
+            'CREATE TABLE IF NOT EXISTS state_events ('
+            'identifier integer primary key, source_statechange_id integer NOT NULL, data binary, '
+            'FOREIGN KEY(source_statechange_id) REFERENCES state_changes(id)'
```

Comment:
Why do we need this foreign key? I'm asking because this requires all the state_changes to be saved in the WAL and also forces a coupling among the state_change code and event code.